File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const n=prices.length
    var answer = new Array(n).fill(0);
    const stack=[]
    
    for(let i=0;i<n;i++){
        while(stack.length>0&&prices[i]<prices[stack[stack.length-1]]){
            const prevIdx=stack.pop();
            answer[prevIdx]=i-prevIdx
        }
        stack.push(i)
    }
    
    while(stack.length>0){
        const prevIdx=stack.pop();
        answer[prevIdx]= n-1-prevIdx
    }
    
    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    l = len(prices)
    
    for i in range(l):
        for j in range(i+1,l):
            if(prices[i]>prices[j]):
                answer.append(j-i)
                break
        if(len(answer)!=i+1):
            answer.append(j-i)
    return answer